### ISSUE TITLE
Comparation between  different types of int causing an error

### ISSUE THREAD

**Issue by JorgeMaker** [matthewelse/pyxpcconnection#7]
While executing: 

sudo python setup.py install

the following message appears causing an error:
 
src/XpcConnectionBase.cpp:72:27: warning: comparison of integers of different signs: 'int' and 'std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> >::size_type' (aka 'unsigned long') [-Wsign-compare]
        for (int i = 0; i < str.size(); i++) {
                        ~ ^ ~~~~~~~~~~
1 warning generated.
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c src/bindings.cpp -o build/temp.macosx-10.9-x86_64-3.7/src/bindings.o -std=c++11 -stdlib=libc++ -mmacosx-version-min=10.9
creating build/lib.macosx-10.9-x86_64-3.7
g++ -bundle -undefined dynamic_lookup -arch x86_64 -g build/temp.macosx-10.9-x86_64-3.7/src/XpcConnectionBase.o build/temp.macosx-10.9-x86_64-3.7/src/bindings.o -lboost_python-py34 -o build/lib.macosx-10.9-x86_64-3.7/xpcconnection.cpython-37m-darwin.so
ld: library not found for -lboost_python-py34
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command 'g++' failed with exit status 1

Any suggestion to solve this issue?

**Comment by matthewelse** [comment]
Looks like the comparison is just a warning, so it's not a huge issue.

I think the real error is that your linker can't find boost python - have you installed it?

**Comment by JorgeMaker** [comment]
I thing so since when I execute " brew install boost-python " I get the following output: 

 _MacBook-Pro-de-Jorge:bleep Jorge$ brew install boost-python
Warning: boost-python 1.67.0 is already installed and up-to-date
To reinstall 1.67.0, run `brew reinstall boost-python`

MacBook-Pro-de-Jorge:PycharmProjects Jorge$ brew install boost-python
Warning: boost-python 1.67.0 is already installed and up-to-date
To reinstall 1.67.0, run `brew reinstall boost-python`_
 
---------------
But in fact what I wanted too do was  was to setup bleep,  but it fails by some unknown reason for me at it shows a warning an the final error at XpcConnectionBase.cpp belonging to pyxpcconnection so i tried to solve this issue. 

The issue with bleep is the following;

_(venv)MacBook-Pro-de-Jorge:PycharmProjects Jorge$ brew install boost-python
Warning: boost-python 1.67.0 is already installed and up-to-date
To reinstall 1.67.0, run `brew reinstall boost-python`
(venv)MacBook-Pro-de-Jorge:PycharmProjects Jorge$ cd bleep
(venv)MacBook-Pro-de-Jorge:bleep Jorge$ sudo python setup.py install_

_src/XpcConnectionBase.cpp:72:27: warning: comparison of integers of different signs: 'int' and 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size_type'
      (aka 'unsigned long') [-Wsign-compare]
        for (int i = 0; i < str.size(); i++) {
                        ~ ^ ~~~~~~~~~~
1 warning generated.
ld: library not found for -lboost_python-py34
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: Setup script exited with error: command 'g++' failed with exit status 1_

------

Do not have any clue how to solve this error. Any help will be very appreciated :)
Thanks in advance for your attention and you very nice python lib. 👍 

PS: I am running on a macOS Mojave 10.14.1 (18B75)

**Comment by danybr-dev** [comment]
I have the same issue!
I'm not able to install pyxpcconnection due to "ld: library not found for -lboost_python-py34".

I tried some fix like  "sudo ln -s /usr/local/lib/libboost_python3.a /usr/local/lib/libboost_python-py34.a" but error doesn't change!

I have macOS Mojave and it is my first install of your libraries.

EDIT: By using this commands I fixed the error of linking that library.

```
ln -s /usr/local/lib/libboost_python37.a /usr/local/lib/libboost_python-py34.a 
ln -s /usr/local/lib/libboost_python37.dylib /usr/local/lib/libboost_python-py34.dylib 

```
Now I have another problem with lstdc++ library:

ld: library not found for -lstdc++

EDIT2:
Solved by using XCode Command Line Tools v9.4.1 

To unistall XCode CLT that you have already installed follow this link:

https://apple.stackexchange.com/questions/308943/how-do-i-uninstall-the-command-line-tools-for-xcode?rq=1
